### ISSUE TITLE
Issues with linkcheck?

### ISSUE THREAD

**Issue by sergisiso** [stfc/PSyclone#2000]
When merging #1970 to master I noticed that the "cd doc/user_guide; make linkcheck" fails not with a link error but with:
```
Exception occurred:
  File "/home/runner/.local/lib/python3.10/site-packages/sphinx/util/nodes.py", line 280, in get_node_line
    raise ValueError("node line not found")
writing output... [ 88%] system_specific_setup
ValueError: node line not found
```
I proceeded with the merge because this PR didn't touch anything related, so I don't think it comes from it. And the previous "cd doc/user_guide; make html" statement in the CI does succeed.
Also if I do "cd doc/user_guide; make linkcheck" in my computer it succeeds, so it may be something specific to github.

If this doesn't get resolved by itself in the next CI runs it should be investigated.

**Comment by sergisiso** [comment]
The issue is with Sphinx 6.1, I pinned the dependency for now and referenced this issue and I leave the issue open until we can unpin it.

**Comment by arporter** [comment]
Thanks @sergisiso. Could you post the link to the Sphinx Issue on this ticket?

**Comment by sergisiso** [comment]
https://www.github.com/sphinx-doc/sphinx/issues/11119

**Comment by hiker** [comment]
Already fixed in sphinx 6.1.3
